### PR TITLE
Catalog: Display badges for Angular plugins and disable install if Angular is disabled

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

When Angular support is disabled in the instance:

It changes some UI elements in the plugins catalog only for Angular plugins:

### Plugins catalog page

immagine

- Adds an "Angular" badge in the plugin card

### Plugin details page

immagine

- Disables the "Install" button. This will also be done for the "Install via grafana.com" button for HG
- Displays an alert with more info and a link to Angular deprecation docs (text to be changed)

**Why do we need this feature?**

Improves Angular deprecation discoverability for operators when browsing the plugins catalog.

**Who is this feature for?**

Grafana operators

**Which issue(s) does this PR fix?**:

Related to #68974

**Special notes for your reviewer:**

Requires the following PRs to be merged in GCOM, so `isAngular` is returned by the API:
- https://github.com/grafana/grafana-com/pull/6539
- https://github.com/grafana/grafana-com/pull/6748

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/plugins/admin/components/InstallControls/ExternallyManagedButton.tsx [hunk 1]
```diff
@@ -35,7 +37,12 @@ export function ExternallyManagedButton({ pluginId, pluginStatus }: ExternallyMa
   }
 
   return (
-    <LinkButton href={externalManageLink} target="_blank" rel="noopener noreferrer">
+    <LinkButton
+      disabled={!config.angularSupportEnabled && angularDetected}
```
Comment (xnyo) [inline public/app/features/plugins/admin/components/InstallControls/ExternallyManagedButton.tsx:41]: TODO: What do we do if the plugin needs to be updated? ("Update via grafana.com" some lines above).

I'm not sure if we can enable/disable depending on whether the target update version is using Angular or not

Comment (xnyo) [reply public/app/features/plugins/admin/components/InstallControls/ExternallyManagedButton.tsx:41]: The same also applies when plugins aren't managed externally (InstallControlButtons.tsx)

Comment (sympatheticmoose) [reply public/app/features/plugins/admin/components/InstallControls/ExternallyManagedButton.tsx:41]: is there a way to permit updates, but disable new installs?

Comment (mckn) [reply public/app/features/plugins/admin/components/InstallControls/ExternallyManagedButton.tsx:41]: Good question, maybe reach out to someone in @grafana/hosted-grafana-team for some input?

File: public/app/features/plugins/admin/components/Badges/PluginUpdateAvailableBadge.tsx [hunk 2]
```diff
@@ -12,13 +12,7 @@ type Props = {
...
   const styles = useStyles2(getStyles);
-
-  // Currently renderer plugins are not supported by the catalog due to complications related to installation / update / uninstall.
-  if (plugin.hasUpdate && !plugin.isCore && plugin.type !== PluginType.renderer) {
-    return <p className={styles.hasUpdate}>Update available!</p>;
-  }
-
-  return null;
+  return <p className={styles.hasUpdate}>Update available!</p>;
 }
```
Comment (mckn) [inline public/app/features/plugins/admin/components/Badges/PluginUpdateAvailableBadge.tsx:16]: I really liked that you moved this from this component to the call site for better overview!

**Review by mckn — COMMENTED** [review]
Code looks really good! Great work on this 🙌🏻 I guess we would need to get some input on how to handle the angular deprecated scenario in cloud.

File: public/app/features/plugins/admin/components/PluginListItemBadges.tsx [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
+  PluginAngularBadge,
+} from './Badges';
 
 type PluginBadgeType = {
   plugin: CatalogPlugin;
 };
 
 export function PluginListItemBadges({ plugin }: PluginBadgeType) {
+  // Currently renderer plugins are not supported by the catalog due to complications related to installation / update / uninstall.
+  const hasUpdate = plugin.hasUpdate && !plugin.isCore && plugin.type !== PluginType.renderer;
```
Comment (mckn) [inline public/app/features/plugins/admin/components/PluginListItemBadges.tsx:22]: I really liked that you moved this from this component to the call site for better overview!

File: public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+import { PluginErrorCode } from '@grafana/data';
+import { Badge } from '@grafana/ui';
+
+type Props = { error?: PluginErrorCode };
+
+export function PluginAngularBadge({ error }: Props): React.ReactElement {
+  return <Badge icon="exclamation-triangle" text="Angular" color="orange" tooltip="Legacy Angular plugin" />;
```
Comment (academo) [inline public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx:9]: IMO the tooltip information here is lacking. not everyone understand what is angular or why is legacy.

What about making the tooltip: `Plugin's use of legacy Angular platform unsupported by Grafana may impact functionality.`

File: public/app/features/plugins/admin/components/Badges/PluginUpdateAvailableBadge.tsx [hunk 5]
```diff
@@ -12,13 +12,7 @@ type Props = {
...
 export function PluginUpdateAvailableBadge({ plugin }: Props): React.ReactElement | null {
   const styles = useStyles2(getStyles);
-
-  // Currently renderer plugins are not supported by the catalog due to complications related to installation / update / uninstall.
-  if (plugin.hasUpdate && !plugin.isCore && plugin.type !== PluginType.renderer) {
-    return <p className={styles.hasUpdate}>Update available!</p>;
-  }
-
-  return null;
+  return <p className={styles.hasUpdate}>Update available!</p>;
```
Comment (academo) [inline public/app/features/plugins/admin/components/Badges/PluginUpdateAvailableBadge.tsx:15]: Is this change intended for this PR? It doesn't look angular related.

File: public/app/features/plugins/admin/components/PluginDetailsAngularDeprecation.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+// An Alert showing information about Angular deprecation notice.
+// If the plugin does not use Angular (!plugin.angularDetected), it returns null.
+export function PluginDetailsAngularDeprecation({ plugin, className }: Props): React.ReactElement | null {
+  if (!plugin.angularDetected) {
+    return null;
+  }
+  return (
+    <Alert severity="warning" title="Angular plugin" className={className}>
+      <p>This plugin is using deprecated plugin APIs.</p>
```
Comment (academo) [inline public/app/features/plugins/admin/components/PluginDetailsAngularDeprecation.tsx:20]: Similar to the tooltip, we need to expand this copy. Explain why is deprecated and what effects installing or enabling angular plugins have.

File: public/app/features/plugins/admin/components/PluginDetailsPage.tsx [hunk 7]
```diff
@@ -73,6 +74,7 @@ export function PluginDetailsPage({
     <Page navId={navId} pageNav={navModel} actions={actions} subTitle={subtitle} info={info}>
       <Page.Contents>
         <TabContent className={styles.tabContent}>
+          <PluginDetailsAngularDeprecation plugin={plugin} className={styles.alert} />
```
Comment (academo) [inline public/app/features/plugins/admin/components/PluginDetailsPage.tsx:77]: nit: do the `if angular` logic here and not inside the component

File: public/app/features/plugins/admin/components/Badges/PluginUpdateAvailableBadge.tsx (hunk 5 above)
Comment (xnyo) [reply public/app/features/plugins/admin/components/Badges/PluginUpdateAvailableBadge.tsx:15]: This isn't angular related but it caused additional spacing to be present when there was no update. By moving the check to the caller and not rendering anything there, the spacing was correct

Comment (xnyo) [reply public/app/features/plugins/admin/components/Badges/PluginUpdateAvailableBadge.tsx:15]:
```
<PluginUpdateAvailableBadge plugin={plugin} />
```

👇 

```
{hasUpdate && <PluginUpdateAvailableBadge plugin={plugin} />}
```

I am referring to this in [PluginListItemBadges.tsx](https://github.com/grafana/grafana/pull/69084/files#diff-2de8bc242744d4ce83055ef3d1105c140cc5d8c735e07dca9b3d80a5953433a8)

File: public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx (hunk 4 above)
Comment (xnyo) [reply public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx:9]: Agree. @sympatheticmoose do you have any other suggestions about copy for this tooltip and https://github.com/grafana/grafana/pull/69084/files#r1227801227 ?

Comment (sympatheticmoose) [reply public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx:9]: For the tooltip, how about something fairly blunt:
> This plugin uses deprecated functionality, support for which is being removed.

Only concern is how well that sentence reads for folks where english isn't their first language. The tooltip we need to keep quite brief, but it is tricky to convey the nuance that it might already be incompatible, and definitely will be incompatible in the future.

Then for the warning text where we can be more verbose:
> This plugin uses a deprecated, legacy platform based on AngularJS and may be incompatible depending on your Grafana configuration.

Comment (sympatheticmoose) [reply public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx:9]: if there is an easy way to detect the value of the config setting for angular support and to dynamically change the text based on that, let me know and I'll rethink :)

Comment (xnyo) [reply public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx]: @sympatheticmoose I have updated the copy as suggested.
And yes, it would be fairly easy to do something like:

```js
This plugin uses a deprecated, legacy platform based on AngularJS and
        {angularSupportEnabled === undefined
          ? ' may be incompatible depending on your Grafana configuration.'
          : angularSupportEnabled
          ? ' will stop working in future releases of Grafana.'
          : ' is incompatible with your current Grafana configuration.'}
```

WDYT?

Comment (sympatheticmoose) [reply public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx]: Nice! I like your suggestion and agree if we want to change the edit panel text that can be a follow up

Comment (xnyo) [reply public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx]: Changed :smiley:

File: public/app/features/plugins/admin/components/InstallControls/ExternallyManagedButton.tsx (hunk 1 above)
Comment (xnyo) [reply public/app/features/plugins/admin/components/InstallControls/ExternallyManagedButton.tsx:41]: I tested the various scenarios and with the current code (always allow update, allow installing only if angular is enabled) and it works as expected.

It's always possible to update plugins when Angular is enabled.
When Angular is disabled, it's possible to "update" (or more correctly, re-install) an old Angular plugin only if it has been updated to React.

I am marking this as resolved.

File: public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import { PluginErrorCode } from '@grafana/data';
+import { Badge } from '@grafana/ui';
+
+type Props = { error?: PluginErrorCode };
+
+export function PluginAngularBadge({ error }: Props): React.ReactElement {
```
Comment (mckn) [inline public/app/features/plugins/admin/components/Badges/PluginAngularBadge.tsx:8]: nit: Looks like we can remove the `Props` type since we are not using the passed `error`.

File: public/app/features/plugins/admin/components/PluginDetailsAngularDeprecation.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  return (
+    <Alert severity="warning" title="Angular plugin" className={className}>
+      <p>
+        This plugin uses a deprecated, legacy platform based on AngularJS and
+        {angularSupportEnabled === undefined
+          ? ' may be incompatible depending on your Grafana configuration.'
+          : angularSupportEnabled
+          ? ' will stop working in future releases of Grafana.'
+          : ' is incompatible with your current Grafana configuration.'}
+      </p>
```
Comment (mckn) [inline public/app/features/plugins/admin/components/PluginDetailsAngularDeprecation.tsx:25]: nit: I would probably put this in an private function with early exits to prevent nesting the ternary operator. But it is completely up to you 🙏🏻

**Review by mckn — APPROVED** [review]
LGTM!

File: public/app/features/plugins/admin/components/PluginDetailsAngularDeprecation.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+      <a
+        href="https://grafana.com/docs/grafana/latest/developers/angular_deprecation/"
+        className="external-link"
+        target="_blank"
+        rel="noreferrer"
+      >
+        Read more about Angular support deprecation.
```
Comment (sympatheticmoose) [inline public/app/features/plugins/admin/components/PluginDetailsAngularDeprecation.tsx:33]: @shih-chris Could you advise what we would need to add here in order to start tracking whether a user from an organization has clicked this link?

**Review by sympatheticmoose — APPROVED** [review]
Great work on this @xnyo
